src/screens/login/LoginScreen.js

```diff
@@ -172,6 +175,55 @@ const LoginSteps = ({
     }))
   }
 
+  const startOidcOAuth = async (fqdn, code) => {
+    if (await NetService.isOffline())
+      NetService.handleOffline(routes.authenticate)
+
+    try {
+      const instance = 'http://' + fqdn
+      const client = await createClient(instance)
+
+      await client.certifyFlagship()
+
+      const result = await getOidcAccessToken(client, code)
+
+      if (result.state === STATE_2FA_NEEDED) {
+        setState(oldState => ({
+          ...oldState,
+          step: TWO_FACTOR_AUTHENTICATION_STEP,
+          isOidc: true,
+          stepReadonly: false,
+          client: result.client,
```

Comment:
Do we really need to put the client in a state? It seems an anti-pattern here

Reply:
> but maybe we already do that and in that case this is not the responsibility of this PR to fix it 

Yes, the entire LoginScreen pattern is based on a local state that keep a `client` between steps and call `setClient` only when everything is done

We tried to switch this behavior for a `useClient()` based one some month ago. But we encountered some problematics with login/logout scenarios, so we delayed this task.